pkg/registry/registry.go

```diff
@@ -255,36 +254,43 @@ func (r *Registry) handleMirror(rw mux.ResponseWriter, req *http.Request, key st
 				return
 			}
 
-			// Modify response returns and error on non 200 status code and NOP error handler skips response writing.
-			// If proxy fails no response is written and it is tried again against a different mirror.
-			// If the response writer has been written to it means that the request was properly proxied.
-			succeeded := false
 			scheme := "http"
 			if req.TLS != nil {
 				scheme = "https"
 			}
-			u := &url.URL{
+			u := url.URL{
 				Scheme: scheme,
 				Host:   ipAddr.String(),
+				Path:   req.URL.Path,
+				// TODO: Should this error early if not set?
+				RawQuery: fmt.Sprintf("ns=%s", req.URL.Query().Get("ns")),
 			}
-			proxy := httputil.NewSingleHostReverseProxy(u)
-			proxy.Transport = r.transport
-			proxy.ErrorHandler = func(_ http.ResponseWriter, _ *http.Request, err error) {
-				log.Error(err, "proxy failed attempting next")
+			forwardReq, err := http.NewRequest(req.Method, u.String(), nil)
```

Comment:
Use NewRequestWithContext and the req.Context()

Reply:
Good catch, totally logical thing to do.